### PR TITLE
Added ecmo to oxygen_therapy attribute

### PR DESCRIPTION
**Pull request by pipliggins**
plus made attribute super-type where necessary
Added mask_prongs attribute to relevant parsers

This only covers parsers that are currently in main, not those still at the pull request stage.

### CODE REVIEW

**Review by abhidg — APPROVED** [review]

File: isaric/parsers/ccp-ghana.toml [hunk 1]
```diff
@@ -824,6 +824,11 @@
     description = "High-flow nasal canula oxygen therapy"
     ref = "Y/N/NK"
 
+  [[visit.treatment_oxygen_therapy.fields]]
+    field = "overall_extracorp_prtrt"
+    description = "Extracorporeal (ECMO) support?"
+    values = { 1 = true, 2 = false }
+
 [visit.treatment_prone_position]
```
Comment (sadiekelly) [inline isaric/parsers/ccp-ghana.toml:832]: @pipliggins variable ecls can be added here also

File: isaric/parsers/ccp-guinea.toml [hunk 2]
```diff
@@ -667,9 +675,11 @@
...
 
 [visit.treatment_noninvasive_ventilation]
-  field = "overall_noninvasive_proccur"
-  description = "Non-invasive ventilation (eg BiPAP, CPAP)?"
-  ref = "Y/N/NK"
+  combinedType = "any"
+  fields = [
+    { field = "overall_noninvasive_proccur", description = "Non-invasive ventilation (eg BiPAP, CPAP)?", ref = "Y/N/NK" },
+    { field = "overall_oxygen_cmtype___5", description = "CPAP/NIV mask", values = { 2 = true } },
+  ]
```
Comment (sadiekelly) [inline isaric/parsers/ccp-guinea.toml:682]: @pipliggins looking back at my initial review, seems there is no data collected in many of these overall_ variables. the variable non_inv_vent can be added, but data is only available for a small number of ICU patients

File: isaric/parsers/ccp-guinea.toml [hunk 3]
```diff
@@ -680,9 +690,14 @@
...
-  description = "Oxygen therapy?"
-  ref = "Y/N/NK"
+  combinedType = "any"
+  fields = [
+    { field = "overall_oxygen_cmoccur", description = "Oxygen therapy?", ref = "Y/N/NK" },
+    { fieldPattern = "overall_oxygen_cmtype___.*", description = "Oxygen interface type", values = { 2 = true } },
+    { field = "overall_invasive_proccur", description = "Invasive ventilation?", ref = "Y/N/NK" },
+    { field = "overall_noninvasive_proccur", description = "Non-invasive ventilation?", ref = "Y/N/NK" },
+    { field = "overall_extracorp_prtrt", description = "Extracorporeal (ECMO) support", ref = "Y/N/NK" },
+  ]
```
Comment (sadiekelly) [inline isaric/parsers/ccp-guinea.toml:701]: @pipliggins the ICU variables can_oxy_ther, non_inv_vent, inv_vent and ecls can be added but not much data provided for these (overall_ variables not completed)

File: isaric/parsers/isaric-ecmo.toml [hunk 4]
```diff
@@ -760,13 +740,18 @@
...
     { field = "daily_fi02_lbyn", ref = "Y/N/NK", description = "Any supplemental oxygen?" },
+    { field = "invasive_proccur", values = { 1 = true, 2 = false }, description = "Invasive ventilation?" },
+    { field = "daily_invasive_prtrt", values = { 1 = true, 2 = false }, description = "Invasive ventilation?" },
+    { field = "nasal_highflow_proccur", ref = "Y/N/NK" },
+    { field = "daily_nasaloxy_cmtrt", ref = "Y/N/NK" },
+    { field = "noninvasive_proccur", values = { 1 = true, 2 = false }, description = "Noninvasive ventilation?" },
+    { field = "daily_noninvasive_prtrt", values = { 1 = true, 2 = false }, description = "Noninvasive ventilation?" },
+    { field = "extracorp_prtrt", ref = "Y/N/NK" },
+    { field = "daily_ecmo_prtrt", ref = "Y/N/NK" },
+    { field = "pronevent_prtrt", description = "Ventilation in prone position?", ref = "Y/N/NK" },
```
Comment (sadiekelly) [inline isaric/parsers/isaric-ecmo.toml:752]: @pipliggins the CRF label for this variable indicates Prone positioning, although the variable name suggests prone ventilation. The follow up duration question also refers to prone positioning, so perhaps it is best to not include this variable in the oxygen_therapy attribute

File: isaric/parsers/isaric-rapid.toml [hunk 5]
```diff
@@ -897,6 +903,15 @@
     { field = "oxygen_cmtype", values = { 2 = true }, description = "Oxygen therapy? (interface)" },
   ]
 
+[visit.treatment_oxygen_mask_prongs]
+  combinedType = "any"
+  fields = [
+    { field = "overall_oxygen_cmtype___1", values = { 1 = true, 0 = false }, description = "Nasal prongs" },
+    { field = "overall_oxygen_cmtype___3", values = { 1 = true, 0 = false }, description = "Mask" },
+    { field = "overall_oxygen_cmtype___4", values = { 1 = true, 0 = false }, description = "Mask with reservoir" },
+    { field = "oxygen_cmtype", values = { 1 = true, 3 = true, 4 = true }, description = "Oxygen therapy? (interface)" },
```
Comment (sadiekelly) [inline isaric/parsers/isaric-rapid.toml:912]: @pipliggins the overall_oxygen_cmtype___ variables are checkboxes so I think best to only map the positive responses (0 responses are default entries where the checkbox was not ticked)

File: isaric/parsers/western-australia.toml [hunk 6]
```diff
@@ -320,9 +320,12 @@
...
   combinedType = "any"
   fields = [
-    { field = "overall_oxygen_cmoccur", ref = "Y/N/NK", description = "Oxygen therapy?" },
-    { field = "daily_oxygen_cmoccur", ref = "Y/N/NK", description = "Oxygen therapy?" },
-    { field = "oxygen_cmoccur", ref = "Y/N/NK", description = "Oxygen therapy on first day of admission?" },
+    { fieldPattern = ".*oxygen_cmoccur", ref = "Y/N/NK", description = "Oxygen therapy?" },
+    { fieldPattern = "overall_oxygen_cmtype.*", ref = "Y/N/NK" },
+    { fieldPattern = ".*oxygen_cmtype", values = { 1 = true, 2 = true, 3 = true, 4 = true, 5 = true } },
+    { fieldPattern = ".*invasive_proccur", ref = "Y/N/NK", description = "Invasive ventilation?" },
+    { fieldPattern = ".*noninvasive_proccur", ref = "Y/N/NK", description = "Invasive ventilation?" },
```
Comment (sadiekelly) [inline isaric/parsers/western-australia.toml:327]: @pipliggins description for row 327 should be Non-invasive ventilation

File: isaric/parsers/western-australia.toml [hunk 7]
```diff
@@ -480,6 +483,15 @@
...
   ]
 
+[visit.treatment_oxygen_mask_prongs]
+  combinedType = "any"
+  fields = [
+    { fieldpattern = ".*oxygen_therapy", values = { 1 = true, 2 = false, 3 = true, 4 = true, 5 = false } },
+    { field = "overall_oxygen_cmtype___1", values = { 0 = false, 1 = true } },
+    { field = "overall_oxygen_cmtype___3", values = { 0 = false, 1 = true } },
+    { field = "overall_oxygen_cmtype___4", values = { 0 = false, 1 = true } },
+  ]
```
Comment (sadiekelly) [inline isaric/parsers/western-australia.toml:493]: @pipliggins should ".*oxygen_therapy" on row 489 be matching field pattern for *_oxygen_cmtype?

for the cmtype___ checkbox variables I think just map the positive responses

**Pull request merged**